tests/e2e/cypress/integration/Autologin/01-autologin/index.ts

```diff
@@ -0,0 +1,171 @@
+  cy.intercept({
+    method: 'GET',
+    url: '/centreon/include/common/userTimezone.php',
+  }).as('getTimeZone');
+});
+
+Given('an Administrator is logged in the platform', () => {
+  return cy
+    .loginByTypeOfUser({ jsonName: 'admin', preserveToken: true })
+    .wait('@getNavigationList')
+    .navigateTo({
+      page: 'Centreon UI',
+      rootItemNumber: 4,
+      subMenu: 'Parameters',
+    })
+    .wait('@getTimeZone')
+    .getIframeBody();
+});
+
+When('the administrator activates autologin on the platform', () => {
+  return cy
+    .getFormFieldByIndex(30)
+    .find('[type="checkbox"]')
+    .check({ force: true })
+    .should('be.checked')
+    .getIframeBody()
+    .find('input[name="submitC"]')
+    .click({ force: true })
+    .reload();
+});
+
+Then(
+  'any user of the plateform should be able to generate an autologin link',
+  () => {
+    return cy
+      .getContainsFromProfileIcon('Edit profile')
+      .click()
+      .visit('/centreon/main.php?p=50104&o=c')
+      .wait('@getTimeZone')
+      .getIframeBody()
+      .find('form #tab1')
+      .within(() => {
+        cy.get('input[name="contact_gen_akey"]').should('be.visible');
+        cy.get('#aKey').invoke('val').should('not.be.undefined');
+      });
+  },
+);
+
+Given(
+  'an authenticated user and autologin configuration menus can be accessed',
+  () => {
+    return cy
+      .logout()
+      .reload()
+      .loginByTypeOfUser({ jsonName: 'user', preserveToken: true })
+      .wait('@getNavigationList')
+      .getContainsFromProfileIcon('Edit profile')
+      .visit('/centreon/main.php?p=50104&o=c')
+      .wait('@getTimeZone')
+      .getIframeBody()
+      .find('form #tab1')
+      .within(() => {
+        cy.get('input[name="contact_gen_akey"]').should('be.visible');
+        cy.get('#aKey').should('be.visible');
+      });
+  },
+);
+
+When('a user generate his autologin key', () => {
+  return cy
+    .getIframeBody()
+    .find('form #tab1 table tbody tr')
+
+    .within(() => {
+      cy.get('input[name="contact_gen_akey"]').click();
+      cy.get('#aKey').invoke('val').should('not.be.undefined');
+    });
+});
+
+Then('the key is properly generated and displayed', () => {
+  return cy
+    .getIframeBody()
+    .find('form #tab1 table tbody tr')
+    .within(() => {
+      cy.get('input[name="contact_autologin_key"]')
+        .invoke('val')
+        .should('not.be.undefined');
+    })
+    .getIframeBody()
+    .find('form input[name="submitC"]')
+    .eq(0)
+    .click()
+    .reload();
+});
+
+Given('a User with autologin key generated', () => {
+  cy.getContainsFromProfileIcon('Copy autologin link').should('be.exist');
+});
+
+When('a User generates an autologin link', () => {
+  return cy
+    .navigateTo({
+      page: 'Templates',
+      rootItemNumber: 2,
+      subMenu: 'Hosts',
+    })
+    .wait('@getTimeZone')
+    .getIframeBody()
+    .find('form')
+    .should('be.exist')
+    .getContainsFromProfileIcon('Copy autologin link')
```

Comment:
You can break the chain after a `click` / `reload` / `should`. It makes the code more readable and easier to maintain.
```suggestion
   cy.getIframeBody().find('form').getContainsFromProfileIcon('Copy autologin link')
```